lib/runner.js

```diff
@@ -302,7 +302,13 @@ Runner.prototype.hook = function(name, fn) {
       }
       if (err) {
         if (err instanceof Pending) {
-          suite.pending = true;
+          if (name === 'beforeEach' || name === 'afterEach') {
+            self.test.pending = true;
+          } else {
+            suite.tests.forEach(function(test) {
+              test.pending = true;
+            });
```

Comment:
setting the suite to be pending, at this point, will cause the `after` hook to not get executed.